apps/emqx_authn/src/emqx_authn_password_hashing.erl

```diff
@@ -0,0 +1,139 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_authn_password_hashing).
```

Comment:
Just to confirm if I understood correctly the goals of this module and `emqx_passwd`:

This one is a more high-level password hashing interface and is recommended for general usage, while `emqx_passwd` has more low-level functions not be used indiscriminately?

(Besides that this module also provides the schema to configure the hashing algorithms)

Reply:
Yes, `emqx_authn_password_hashing` belongs to `emqx_authn` app and makes a single interface for working with password hashing in authn backends.
This helps to avoid schema and code dupliation across authn backends and should reduce possibility of errors.